tonic_validate/validate_scorer.py

```diff
@@ -56,20 +66,44 @@ def __init__(
     def _score_item_rundata(self, response: LLMResponse) -> RunData:
         scores: Dict[str, Union[float, None]] = {}
         # We cache per response, so we need to create a new OpenAIService
-        openai_service = OpenAIService(self.encoder, self.model_evaluator)
+        openai_service = OpenAIService(
+            self.encoder, self.model_evaluator, max_retries=self.max_llm_retries
+        )
         for metric in self.metrics:
-            try:
-                score = metric.score(response, openai_service)
-            except Exception as e:
-                if not self.fail_on_error:
-                    score = None
-                    logger.error(
-                        f"Error calculating score for {metric.name}, setting score to None."
+            tries = 0
+            exceptions = []
+            last_cache = copy.deepcopy(openai_service.cache)
+            while tries < 3:
```

Comment:
Should this be `while tries < self.max_scoring_retries`?

Reply:
I think I accidentally left that in. My bad